consensus/istanbul/types.go

```diff
@@ -914,7 +914,12 @@ func NewVersionCertificate(version uint, signingFn func([]byte) ([]byte, error))
 	if err != nil {
 		return nil, err
 	}
-	return vc, vc.recoverAddressAndPubKey()
+	err = vc.recoverAddressAndPubKey()
+	if err != nil {
+		return nil, err
+	}
+
+	return vc, nil
```

Comment:
This change is not related to this PR's goal, it was made because @oneeman felt that clarity was harmed by having a return statement call a function (`recoverAddressAndPubKey`) that modifies the first returned parameter.